src/internal/util/TimeoutError.ts

```diff
@@ -8,7 +8,7 @@
 export class TimeoutError extends Error {
   constructor() {
     super('Timeout has occurred');
-
+    this.name = 'TimeoutError';

```

Comment:
it would be nice if this was declared as a property so it is a visible part of the API contract:
```ts
public readonly name: 'TimeoutError' = 'TimeoutError'
```